features/react_native.feature

```diff
@@ -11,3 +11,80 @@ Scenario: React Native sends requests
     And 1 requests are valid for the android mapping API and match the following:
       | versionCode | versionName | appId                     |
       | 1           | 1.0         | com.bugsnag.android.rnapp |
+
+    And 1 requests are valid for the JS source map API and match the following:
+        | appVersionCode | appVersion | platform |
+        | 1              | 1.0        | android  |
+
+Scenario: Source maps are uploaded when bundling an app with the default project structure
+    When I build the React Native app
+    And I wait to receive 3 requests
+
+    Then 1 requests are valid for the build API and match the following:
+        | appVersionCode | appVersion | buildTool      |
+        | 1              | 1.0        | gradle-android |
+
+    And 1 requests are valid for the android mapping API and match the following:
+        | versionCode | versionName | appId                     |
+        | 1           | 1.0         | com.bugsnag.android.rnapp |
+
+    And 1 requests are valid for the JS source map API and match the following:
+        | appVersionCode | appVersion | platform |
+        | 1              | 1.0        | android  |
+
+Scenario: Source maps are uploaded when assembling an app which uses productFlavors
+    When I set environment variable "USE_RN_FLAVORS" to "true"
+    When I build the React Native app
+    And I wait to receive 6 requests
+
+    Then 2 requests are valid for the build API and match the following:
+        | appVersionCode | appVersion | buildTool      |
+        | 1              | 1.0        | gradle-android |
+        | 1              | 1.0        | gradle-android |
+
+    And 2 requests are valid for the android mapping API and match the following:
+        | versionCode | versionName | appId                         |
+        | 1           | 1.0         | com.bugsnag.android.rnapp.foo |
+        | 1           | 1.0         | com.bugsnag.android.rnapp.bar |
+
+    And 2 requests are valid for the JS source map API and match the following:
+        | appVersionCode | appVersion | platform |
+        | 1              | 1.0        | android  |
+        | 1              | 1.0        | android  |
+
+Scenario: Setting uploadReactNativeMappings to false will prevent any source map upload
+    When I set environment variable "UPLOAD_RN_MAPPINGS" to "false"
```

Comment:
Yeah, related to my earlier query about Hermes - this use looks like it would be fine, but my concern is that it might lead people to use `RN_ENABLE_HERMES` incorrectly over time.  I.e. assuming setting it to "false" would mean "not enabled".